core/shared/src/main/scala/io/chrisdavenport/rediculous/RedisConnection.scala

```diff
@@ -671,10 +671,4 @@ object RedisConnection{
   private def raceNThrowFirst[F[_]: Concurrent, A](nel: NonEmptyList[F[A]]): F[A] = 
     Stream(Stream.emits(nel.toList).evalMap(identity)).covary[F].parJoinUnbounded.take(1).compile.lastOrError
 
-  private class TimeoutConnection[F[_]: Temporal](rC: RedisConnection[F], duration: Duration) extends RedisConnection[F] {
-
-    def runRequest(inputs: Chunk[NonEmptyList[ByteVector]], key: Option[ByteVector]): F[Chunk[Resp]] =
-      rC.runRequest(inputs, key).timeout(duration)
```

Comment:
Sorry if I missed it. What replaces this `.timeout(...)` call? I don't think I saw any new `.timeout(...)` calls introduced above.

Reply:
Oops, you are right, will try to fix it.

Reply:
I took a quick look, and have a naive solution, which is add `timeout` to `request` function of each `Connection` class. Would it work?


```diff
M core/shared/src/main/scala/io/chrisdavenport/rediculous/RedisConnection.scala
@@ -40,7 +40,7 @@ object RedisConnection{
       }
     } 
   }
-  private[rediculous] case class PooledConnection[F[_]: Concurrent](
+  private[rediculous] case class PooledConnection[F[_]: Temporal](
     pool: KeyPool[F, Unit, Socket[F]], timeout: Duration
   ) extends RedisConnection[F]{
     def runRequest(inputs: Chunk[NonEmptyList[ByteVector]], key: Option[ByteVector]): F[Chunk[Resp]] = {
@@ -52,6 +52,7 @@ object RedisConnection{
           case _ => Applicative[F].unit
         }
       }.rethrow
+      .timeout(timeout)
     }
   }
 
```

Reply:
Yes, that's what I've done now. I've also sealed the `RedisConnection` trait to ensure exhaustive matches in `RedisPubSub`. Or is that going too far?

Reply:
> I've also sealed the `RedisConnection` trait to ensure exhaustive matches in `RedisPubSub`. Or is that going too far?

I think that's a good idea.